wordpress-mobile/WordPress-iOS
# Jetpack: remove add site option

Fixes #16210

This PR removes the option to create sites in the Jetpack target.

| Jetpack | WordPress |
| -------- | ----------- |
| ![Simulator Screen Shot - iPhone 11 - 2021-04-01 at 17 19 19](https://user-images.githubusercontent.com/7040243/113349943-289b7000-930f-11eb-8aac-e4c8f0123ddb.png) | ![Simulator Screen Shot - iPhone 11 - 2021-04-01 at 17 24 39](https://user-images.githubusercontent.com/7040243/113349969-318c4180-930f-11eb-959e-bd30d559395f.png) |

### To test

Jetpack

1. Tap the site selector
1. Make sure no "+" button is available

WordPress

1. Tap the site selector
1. Make sure the "+" button is available

## Additional info

I created a new `AppBehavior` file. I think this is better than checking `isJetpack` because it allows the configuration to be generic and feature-specific. This will probably add more code but it's more descriptive and it's not Jetpack-implementation related, we can use it for a lot of other things in the future.

Let me know what do you think, if we want to move ahead with this we can remove the `AppConfiguration` and refactor the parts that uses it.

## Regression Notes
1. Potential unintended areas of impact
WordPress.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I tested WordPress and the behavior there is still the same.

3. What automated tests I added (or what prevented me from doing so)
Visual change, no automated tests added.

PR submission checklist:
